grammar/vernacextend.mlp

```diff
@@ -162,22 +166,27 @@ EXTEND
       (which otherwise could have been another argument) is not passed
       to the VernacExtend interpreter function to discriminate between
       the clauses. *)
-
-  (* ejga: Due to the LocalityFixme abomination we cannot eta-expand
-      [e] as we'd like to, so we need to use the below mess with [fun
-      st -> st].
-
-      At some point We should solve the mess and extend
-      vernacextend.mlp with locality info. *)
   rule:
     [ [ "["; s = STRING; l = LIST0 args; "]";
         d = OPT deprecation; c = OPT classifier; "->"; "["; e = Pcaml.expr; "]" ->
       let () = if s = "" then failwith "Command name is empty." in
-      let b = <:expr< fun loc -> ( let () = $e$ in fun st -> st ) >> in
+      let b = <:expr< fun ~atts ~st -> ( let () = $e$ in st ) >> in
```

Comment:
What is that piece of code below supposed to do? It doesn't seem to be related to localities.

Reply:
You mean `fun_rule`, this is basically duplicate code, `rule` is an adaptor for imperative extensions, `fun_rule` is an adaptor for functional extension. That is to say, they are identical but for:
```
fun ~atts ~st -> ( let () = $e$ in st )
vs
$e$
```
[so `e` in the second case has type `unit` vs in the first which is a functional]

Help on how to better organize this is very welcome, my camlp5-fu is nul.